### PR TITLE
update SDK to track2 in ensureGatewayCreate

### PR DESCRIPTION
**Pull request by bitoku**
### Which issue this PR addresses:

Fixes a part of https://issues.redhat.com/browse/ARO-4665 and https://issues.redhat.com/browse/ARO-7316

This is blocked until https://github.com/Azure/ARO-RP/pull/3638 is released.

### What this PR does / why we need it:

This PR makes `ensureGateway` action use track2 SDK.

### Test plan for issue:

unittest for ensureGateway.
e2e

### Is there any documentation that needs to be updated for this PR?

tech debt cleanup N/A

### How do you know this will function as expected in production? 

cluster installation test

### CODE REVIEW

**Comment by bitoku** [comment]
Waiting for https://github.com/Azure/ARO-RP/pull/3579 to be released because it had a bug about tenant ID settings, and want to confirm it works first.

**Review by mociarain — APPROVED** [review]
LGTM

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -80,7 +80,6 @@ type manager struct {
 	armPublicIPAddresses     armnetwork.PublicIPAddressesClient
 	loadBalancers            network.LoadBalancersClient // TODO: use armLoadBalancers instead.
 	armLoadBalancers         armnetwork.LoadBalancersClient
-	privateEndpoints         network.PrivateEndpointsClient // TODO: use armPrivateEndpoints instead.
```
Comment (tiguelu) [inline pkg/cluster/cluster.go:83]: Can we remove /pkg/util/azureclient/mgmt/network/privateendpoints.go ?

Seems it was only referenced here.

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -94,7 +93,6 @@ type manager struct {
 	denyAssignments          authorization.DenyAssignmentClient
 	fpPrivateEndpoints       network.PrivateEndpointsClient // TODO: use armFPPrivateEndpoints instead.
 	armFPPrivateEndpoints    armnetwork.PrivateEndpointsClient
-	rpPrivateLinkServices    network.PrivateLinkServicesClient // TODO: use armRPPrivateLinkServices instead.
```
Comment (tiguelu) [inline pkg/cluster/cluster.go:97]: Can this be removed too? /pkg/util/azureclient/mgmt/network/privatelinkservices.go

File: pkg/cluster/ipaddresses_test.go [hunk 3]
```diff
@@ -707,7 +707,7 @@ func TestEnsureGatewayCreate(t *testing.T) {
 
 	for _, tt := range []struct {
 		name                     string
-		mocks                    func(*mock_env.MockInterface, *mock_network.MockPrivateEndpointsClient, *mock_network.MockPrivateLinkServicesClient)
```
Comment (tiguelu) [inline pkg/cluster/ipaddresses_test.go:710]: The old mock seem to have been used only here. Can it be removed too?

**Review by tiguelu — APPROVED** [review]
LGTM. Thank you for updating these clients, Ayato.

I left some minor questions, but they can be tackled in a follow-up, if needed.

File: pkg/cluster/cluster.go (hunk 1 above)
Comment (bitoku) [reply pkg/cluster/cluster.go:83]: Not yet. `fpPrivateEndpoints` still uses the old one.

File: pkg/cluster/cluster.go (hunk 2 above)
Comment (bitoku) [reply pkg/cluster/cluster.go:97]: Not yet. pkg/util/purge/purge.go still uses the old one.

**Pull request merged**